file-mon/src/main/java/com/wipro/ats/bdre/filemon/FileMonitor.java

```diff
@@ -29,8 +28,10 @@
 public class FileMonitor implements FileListener {
     private static final Logger LOGGER = Logger.getLogger(FileMonRunnableMain.class);
     private static FileMonitor fileMonitor = null;
-    private SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss");
 
+    private String monDir = null;
+    private String archiveDirName = null;
+    private String filePattern = null;
     public static synchronized FileCopyInfo getFileInfoFromQueue() {
         String key = FileMonitor.fileToCopyMap.firstKey();
         FileCopyInfo fileCopyInfo=fileToCopyMap.remove(key);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "fileCopyInfo". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1488)